src/main/java/com/it4045/pokemonviewer/enterprise/service/PokemonServiceStub.java

```diff
@@ -17,9 +18,19 @@
      */
     @Override
     public Pokemon fetchByName(String pokemonName) {
-        pokemon.setPokemonName("Charizard");
-        pokemon.setPokemonId(1);
-        return pokemon;
+        try {
+            for (int i = 0; i < pokemonList.size(); i++) {
+                if (pokemon.getPokemonName().equals(pokemonName)) {
+                    return pokemon;
+                }
+            }
+        } catch(NullPointerException ex) {
+            throw new NullPointerException(ex.getMessage());
+        }
+        return null;
+//      pokemon.setPokemonName("Charizard");
+//      pokemon.setPokemonId(1);
+//      return pokemon;
```

Comment:
Avoid committing commented-out code to version control, as version control offers better approaches:
1) If you are curious what used to be there, diff commits to see history.
2) If it is an experimental feature, use a feature branch to keep it separate from master.